src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -0,0 +1,615 @@
+        spdxRelationshipAndLicenseElementsToAddToSBOM.Add(licenseDeclaredElement);
+
+        var spdxRelationshipLicenseDeclaredElement = new Spdx30Relationship
+        {
+            From = spdxPackage.SpdxId,
+            RelationshipType = RelationshipType.HAS_DECLARED_LICENSE,
+            To = new List<string> { licenseDeclaredElement.SpdxId }
+        };
+
+        spdxRelationshipLicenseDeclaredElement.AddSpdxId();
+        spdxRelationshipAndLicenseElementsToAddToSBOM.Add(spdxRelationshipLicenseDeclaredElement);
+
+        return spdxRelationshipAndLicenseElementsToAddToSBOM;
+    }
+
+    private Element GenerateLicenseElement(string licenseInfo)
+    {
+        Element licenseElement = null;
+        if (licenseInfo == null)
+        {
+            licenseElement = new NoAssertionElement();
+        }
+        else
+        {
+            licenseElement = new AnyLicenseInfo { Name = licenseInfo };
+        }
+
+        licenseElement.AddSpdxId();
+        return licenseElement;
+    }
+
+    /// <summary>
+    /// Convert Extensions.Entities.RelationshipType to SPDX 3.0 RelationshipType.
+    /// </summary>
+    /// <param name="relationshipType"></param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    private RelationshipType GetSPDXRelationshipType(Extensions.Entities.RelationshipType relationshipType)
+    {
+        switch (relationshipType)
+        {
+            case Extensions.Entities.RelationshipType.CONTAINS: return RelationshipType.CONTAINS;
+            case Extensions.Entities.RelationshipType.DEPENDS_ON: return RelationshipType.DEPENDS_ON;
+            case Extensions.Entities.RelationshipType.DESCRIBES: return RelationshipType.DESCRIBES;
+            case Extensions.Entities.RelationshipType.PREREQUISITE_FOR: return RelationshipType.HAS_PREREQUISITE;
+            case Extensions.Entities.RelationshipType.DESCRIBED_BY: return RelationshipType.DESCRIBES;
+            case Extensions.Entities.RelationshipType.PATCH_FOR: return RelationshipType.PATCHED_BY;
+            default:
+                throw new NotImplementedException($"The spdxRelationship {relationshipType} is currently not " +
+                                                  $"mapped to any SPDX 3.0 spdxRelationship type.");
+        }
+    }
+
+    // Throws a <see cref="MissingHashValueException"/> if the filehashes are missing
+    // any of the required hashes
+    private void EnsureRequiredHashesPresent(Checksum[] fileHashes)
+    {
+        foreach (var hashAlgorithmName in from hashAlgorithmName in RequiredHashAlgorithms
+                                          where !fileHashes.Select(fh => fh.Algorithm).Contains(hashAlgorithmName)
+                                          select hashAlgorithmName)
+        {
+            throw new MissingHashValueException($"The hash value for algorithm {hashAlgorithmName} is missing from {nameof(fileHashes)}");
+        }
+    }
```

Comment:
I moved these methods to a new class. I can't move the required hashes array since it needs to be implemented as part of the IManifestGenerator in the generator files